packages/core/src/middleware/koa-log-session.test.ts

```diff
@@ -0,0 +1,66 @@
+import { Provider } from 'oidc-provider';
+
+import { WithLogContext } from '@/middleware/koa-log';
+import koaLogSession from '@/middleware/koa-log-session';
+import { createContextWithRouteParameters } from '@/utils/test-utils';
+
+const interactionDetails: jest.MockedFunction<() => Promise<unknown>> = jest.fn(async () => ({}));
+
+jest.mock('oidc-provider', () => ({
+  Provider: jest.fn(() => ({
+    interactionDetails,
+  })),
+}));
+
+describe('koaLogSession', () => {
+  const sessionId = 'sessionId';
+  const applicationId = 'applicationId';
+  const logSession = jest.fn();
+  const log = jest.fn();
+  const next = jest.fn();
+
+  interactionDetails.mockResolvedValue({
+    jti: sessionId,
+    params: {
+      client_id: applicationId,
+    },
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should get session info from the provider', async () => {
+    const ctx: WithLogContext<ReturnType<typeof createContextWithRouteParameters>> = {
+      ...createContextWithRouteParameters(),
+      logSession,
+      log,
+    };
+
+    await expect(koaLogSession(new Provider(''))(ctx, next)).resolves.not.toThrow();
+    expect(interactionDetails).toHaveBeenCalled();
+    expect(logSession).toHaveBeenCalledWith({ sessionId, applicationId });
+  });
+
+  it('should log session id and application id', async () => {
```

Comment:
what's the difference between this case and the case in line 33.

Reply:
i should remove `expect(logSession).toHaveBeenCalledWith({ sessionId, applicationId });` from previous case.

updated. PTAL